### ISSUE TITLE
Code for training the ResNet18 w/o extra data?

### ISSUE THREAD

**Issue by kaijiezhu11** [P2333/SCORE#2]
Dear author,

I was wondering if you could kindly guide me on how to train a ResNet18 model without using any additional data, as described in your paper's Section 5.2 and Table 2. I have attempted to locate the relevant code in your released materials, but unfortunately, I have not been able to find it. Additionally, I was unable to locate the code for PGD-AT with SE. I would be most grateful if you could kindly assist me with these queries.

Thank you for your time and consideration.

**Comment by P2333** [comment]
For Section 5.2 and Table 2, we follow the settings and code of [Pang et al. 2021](https://github.com/P2333/Bag-of-Tricks-for-AT), and we substitute the KL divergence with SE.

**Comment by kaijiezhu11** [comment]
Thank you so much for your prompt reply to my issue. 

I implement the PGD-AT with SCORE by replacing the CE loss to LSE loss:

>   y_onehot = F.one_hot(y, num_classes=args.num_classes)
>   robust_output = model(adv_input)
>   logits_adv = F.softmax(robust_output, dim=1)
>   robust_loss = torch.sum((logits_adv - y_onehot) ** 2, dim=-1).mean()
>   opt.zero_grad()
>   robust_loss.backward()
 
I would also like to ask for your guidance on one more thing. Should the CE loss in PGD process be replaced as well? I noticed that in your code (trades.py/trades_loss_LSE), it is replaced.

Thank you again for your help. I appreciate your time and expertise.

**Comment by P2333** [comment]
Yes the CE loss in PGD should be replaced as well. This is because the objective of AT is a min-max problem, i.e., the minimization and maximization share the same objective function.

**Comment by kaijiezhu11** [comment]
> Yes the CE loss in PGD should be replaced as well. This is because the objective of AT is a min-max problem, i.e., the minimization and maximization share the same objective function.

I've tried for both, 

the results are as follows:
Method                                         Std Acc            Robust Acc
PGD-AT                                        81.46                 53.62
PGD-AT SCORE (Not replaced) 81.88                 53.99
PGD-AT SCORE (replaced)        80.04                 53.19


It seems that the CE loss in PGD should **not** be replaced? 


I have been using BN mode as train() when using PGD. Could this be the reason why PGD-AT SCORE (replaced) is not performing as well as PGD-AT SCORE (not replaced)?

**Comment by kaijiezhu11** [comment]
I tried to train PGD-AT & SCORE on CIFAR100, the results worse than vanilla PGD-AT.

Method Std Acc Robust Acc

PGD-AT 55.30 29.50
PGD-AT SCORE 29.28 22.32
PGD-AT SCORE w/o grad clip 36.46 25.20
PGD-AT SCORE (Not replaced) 42.96 27.01

It seems that the the CE loss in PGD should not be replaced as well?

Have you tried to run PGD-AT SCORE with CIFAR100? I wonder if I wrote the wrong code, but it seems not possible since for CIFAR10 it is right.
